### DISCUSSION TITLE
Apexcharts: Set data_generator to only import specific hours

### DISCUSSION THREAD

**Discussion by airman84** [mampfes/ha_epex_spot#53] [Q&A]
Hi guys,

does anyone have any idea, how to set the Apexcharts data_generator to import only hours -2h until +32h?
I know, I can adjust the graph_span and offset, which shows me the data I want, but unfortuntely the scaling in Apexcharts is wrong then.

Here the y-axis is up to 32cent, while the max shown data value is just 26cent. Thats becaue there is data close to 32cent in the data generator, that is filtered out by the graph_span and offset but still affecting the max displayed value on y-axis. By limiting the data_importer to just import the timeframe I want to display, this could be worked around.

Current:
![image](https://github.com/mampfes/ha_epex_spot/assets/73861373/0820e1d1-86a5-43c7-9fe7-1999f312ba32)

Wanted:
![image](https://github.com/mampfes/ha_epex_spot/assets/73861373/876e2b1e-9243-42af-99b6-0492b77d7102)

data_generator: >
        return entity.attributes.data.map((entry, index) => { return [new
        Date(entry.start_time).getTime(), entry.price_ct_per_kwh]; });   

Thank you
Oliver

**Comment by deltamelter** [comment]
Not sure exactly what you mean by only importing -2h until +32h
You can use a filtermap to limit the data displayed, but graph_span and offset are the best way limit the x-axis displayed
If the y-axis doesn't play along, use the max property under yaxis
```
graph_span: 48h
span:
  start: hour
  offset: '-10h'
  ```
```
yaxis:
  - id: gb30
    decimals: 3
    apex_config:
      tickAmount: 20
    max: '|+5|'
    min: 0
```
This takes data all the way to the top but adds a 5 unit gap

**Reply by airman84** [reply]
The EPEX Net price sensor contains here 48h of data. The y-axis is  from 12-27cent, which fits to the displayed data.
```
    graph_span: 48h
    span:
      start: day
```
![image](https://github.com/mampfes/ha_epex_spot/assets/73861373/089c9309-9ce4-4335-9b0c-4c445ebb49e3)

But when changing the values like below, as I am not so interested in the past prices, y-axis does not change, to fit the changed interval of data displayed. I would expect now to change the y-avis to 15-27cent, but it remains with 12-27cent, due to the lowest value of 12,9cent, which is loaded via the data_generator, but not displayed anymore. Min/Max values of apexcharts, seem to belong to Min/Max of available data, but not of displayed data. So my approach would have been, to configure the data_generator, to import the same hours I want to display, so the Min/Max values will fit. 
```
    graph_span: 34h
    span:
      start: hour
      offset: '-2hr'
```
![image](https://github.com/mampfes/ha_epex_spot/assets/73861373/72b4abf0-92e9-429e-9b1b-40dd4db800c0)

**Reply by deltamelter** [reply]
then use filtermap.  It will then only hold the window of data that you are interested in.

**Reply by deltamelter** [reply]
![image](https://github.com/mampfes/ha_epex_spot/assets/27061628/5007626b-f122-4e00-a0e9-7af3cd818d20)

**Reply by airman84** [reply]
Hi,
many thanks for your reply. I would be happy to try it, but the code is cut on the right side. Could you maybe post the code here?
Thank you

**Reply by airman84** [reply]
I just noticed, that I made a mistake when configuring the yaxis in apexcharts... everything works as expected, no need for a changed data_generator. sorry for wasting your time :(
